nexus/auth/src/authz/context.rs

```diff
@@ -196,17 +196,46 @@ pub trait AuthorizedResource: oso::ToPolar + Send + Sync + 'static {
 mod test {
     use crate::authn;
     use crate::authz::Action;
+    use crate::authz::AnyActor;
     use crate::authz::Authz;
     use crate::authz::Context;
-    use crate::db::DataStore;
-    use nexus_test_utils::db::test_setup_database;
+    use crate::authz::RoleSet;
+    use crate::context::OpContext;
     use omicron_test_utils::dev;
     use std::sync::Arc;
 
+    use nexus_db_model::IdentityType;
+    use nexus_db_model::RoleAssignment;
+    use omicron_common::api::external::Error;
+    use omicron_common::api::external::ResourceType;
+    use uuid::Uuid;
+
+    struct FakeStorage {}
+
+    impl FakeStorage {
+        fn new() -> Arc<dyn crate::storage::Storage> {
+            Arc::new(Self {})
+        }
+    }
+
+    #[async_trait::async_trait]
+    impl crate::storage::Storage for FakeStorage {
+        async fn role_asgn_list_for(
+            &self,
+            _opctx: &OpContext,
+            _identity_type: IdentityType,
+            _identity_id: Uuid,
+            _resource_type: ResourceType,
+            _resource_id: Uuid,
+        ) -> Result<Vec<RoleAssignment>, Error> {
+            todo!();
```

Comment:
perhaps we should change this to something like 
```suggestion
            unimplemented!("this test shouldn't need to actually access the database");
```
or something, so that if someone introduces a change that invalidates this assumption, it's more obvious what went wrong?

Reply:
Sounds good, updated!